### PR TITLE
feat: Validate docName and seriesInfo value for I-D

### LINKED ISSUE #1115: Check for draft name divergence
### Description

See https://github.com/ietf-tools/datatracker/issues/7217

Add a validation step in `xml2rfc` to check for draft name divergence in I-Ds for `//rfc/@docname` and `//seriesInfo[@name='Internet-Draft]/@value`.

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by kesara**
* refactor: Remove an obsolete warning
* refactor: Remove redundant import
* feat: Validate docName and seriesInfo value for I-D

Fixes #1115

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -697,6 +697,67 @@ def test_render_reference(self):
...
+        # RFC
+        rfc = lxml.etree.fromstring('''
+<rfc
+    number="9280"
+    docName = "draft-ietf-foobar-23"
+    ipr="trust200902"
+    submissionType="editorial"
+    category="info">
+    <front>
+        <seriesInfo name="Internet-Draft" value="draft-ietf-foobar-23" />
```
Comment (rjsparks) [inline test.py:726]: RFCs don't have this - they have an rfc series info entry instead

File: test.py [hunk 2]
```diff
@@ -697,6 +697,67 @@ def test_render_reference(self):
...
+
+    def test_validate_draft_name(self):
+        # Draft with only docName
+        self.assertTrue(self.writer.validate_draft_name())
+
+        # RFC
+        rfc = lxml.etree.fromstring('''
+<rfc
+    number="9280"
+    docName = "draft-ietf-foobar-23"
```
Comment (rjsparks) [inline test.py:721]: no action needed now, but this will be a place to touch when RSWG approves the change to how docName should be used.

**Review by rjsparks — CHANGES_REQUESTED** [review]
A test detail change requested.

File: test.py (hunk 1 above)
Comment (kesara) [reply test.py]: Updated.

File: test.py (hunk 2 above)
Comment (kesara) [reply test.py]: Update this to look more compliant. :)

**Review by rjsparks — APPROVED** [review]

**Review by rjsparks — APPROVED** [review]

**Pull request merged**